prow/cmd/deck/main_test.go

```diff
@@ -1189,6 +1189,14 @@ func TestHttpStatusForError(t *testing.T) {
 			},
 			expectedStatus: http.StatusGone,
 		},
+		{
```

Comment:
The current way of retrieving http response code always return 5XX for any wrapped errors, see this test case

Reply:
Shouldn't we fix that then?

Reply:
Agree that's the best approach, thought about it for a few minutes but didn't have a good way. Will update here if I can find a better way. Also let me know if you have an idea

Reply:
```
diff --git a/prow/cmd/deck/main.go b/prow/cmd/deck/main.go
index dec42e1b0e..94ad16678c 100644
--- a/prow/cmd/deck/main.go
+++ b/prow/cmd/deck/main.go
@@ -1651,7 +1651,8 @@ type httpError struct {
 }
 
 func httpStatusForError(e error) int {
-       if httpErr, ok := e.(httpError); ok {
+       var httpErr httpError
+       if ok := errors.As(e, &httpErr); ok {
                return httpErr.statusCode
        }
        return http.StatusInternalServerErro
```

Reply:
Awesome, good to learn! I'll give it a shot

Reply:
Done. And btw replaced a few `%v` to `%w`

Reply:
Also updated both PR title and commit msg

Reply:
Thanks. As an aside, we have verification in our downstream repo to enforce error wrapping: https://github.com/openshift/ci-tools/blob/9f710b04bc1319b5374ea7ef26a8dd1e8dd97f8b/hack/lint.sh#L24

And also something to fix it up automatically that works okayish: https://github.com/openshift/ci-tools/blob/9f710b04bc1319b5374ea7ef26a8dd1e8dd97f8b/hack/fixup-error-wrapping.sh#L10

However, I do not know how to integrate new stuff with bazel :upside_down_face: